_posts/2021-05-27-TC-Guard.md

```diff
@@ -0,0 +1,160 @@
+
+### Finished Product
+
+<p style="display:flex; justify-content:space-around">
+  <img src = "/img/2021-05-27-TC-Guard/main.png" style="width: 45vw; max-width: 350px;"> 
+  <img src = "/img/2021-05-27-TC-Guard/dashboard.png" style="width: 45vw; max-width: 350px;">
+</p>
+
+# Kubernetes
+
+<img src = "/img/2021-05-27-TC-Guard/schema.png" class = "image fit">
+
+The figure above shows the block diagram of this project.
+This diagram represents all the components of this internship.
+Each component will be briefly discussed below.
+All applications required for this internship run within a single Kubernetes cluster.
+Inside this cluster, Argo CD is used to keep everything up-to-date.
+The Android application is connected to the TC Guard API using Contour.
+This is also used to expose the Graylog dashboard to the public.
+TC Guard and Graylog are connected so that the logs from TC Guard can be sent to Graylog.
+The data collected by the Android application is stored in the PostgreSQL database via the TC Guard API.
+When the application needs this data, it is retrieved again via the TC Guard API.
+The last block connected to the TC Guard API is Prometheus.
+Prometheus is used to observe the API.
+The resulting data is then displayed in the Grafana dashboard.
+
+### Argo CD
+Argo CD is a Continuous Delivery GitOps tool for Kubernetes.
+Within the internship assignment, this is used to install helm charts on the Kubernetes cluster.
+If a change is made on the Azure DevOps repository, Argo CD will detect this and implement the changes on the Kubernetes cluster via a rolling update.
+
+It was quite satisfying to observe the changes being applied.
+Argo CD made managing our applications way easier.
+For example when we updated Graylog the only thing we had to do was change the version number in a .yaml file and Argo CD took care of everything else.
+
+<img src = "/img/2021-05-27-TC-Guard/argocd.png" class = "image fit">
+
+### Ingress
+Ingress provides routing rules to manage remote users' access to services in a Kubernetes cluster. With Ingress, simple rules can be set for routing traffic without exposing every service on the endpoint.
+While setting up Ingress, a few issues surfaced due to our inexperience with Kubernetes.
+Because the online documentation of Ingress nginx was rather limited, setting up Ingress took a fair amount of time.
+Afterwards, Ingress nginx was replaced by Contour, by this time we had more experience.
+Because of this installation went smoothy.
+
+### Graylog
+The centralized logging system aims to obtain the logs of the API and the Android application in a central place so that it would be easier to detect errors.
+To obtain and view the logs in a central place, Graylog is used.
+We chose Graylog because it is much easier to use than Kibana.
+We knew that Kibana has more features than Graylog, but for this project it was not required.
+
+<img src = "/img/2021-05-27-TC-Guard/graylog.png" class = "image fit">
+
+### Grafana
```

Comment:
This section is not clear. Try define better what Prometheus does and what grafana does. And explain a bit more about issues you had. For people who don't know Prometheus in detail, this part is not easy to understand